src/transformations.jl

```diff
@@ -8,24 +8,32 @@ abstract type AbstractDataTransform end
 
 Apply transformation `t` to vector or matrix `x` in place.
 """
-function transform!(t::AbstractDataTransform, x::AbstractVecOrMat{<:Real})
-    if t.dims == 1
-        return transform!(x, t, x)
-    elseif t.dims == 2
-        return transform!(x', t, x')'
-    end
+transform!(t::AbstractDataTransform, x::AbstractMatrix{<:Real}) = _transform!(x, t, x)
+
+function transform!(t::AbstractDataTransform, x::AbstractVector{<:Real})
+    x = reshape(x, :, 1)
```

Comment:
Use another name than `x` (e.g. `xmat`), so that you can return the original `x` in the end; that's better for type stability anyway. Or simpler, just do `_transform!(x, t, reshape(x, :, 1))` directly.

Reply:
OK, now I totally understandwhat you said.